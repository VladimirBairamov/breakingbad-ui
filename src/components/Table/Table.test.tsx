import React from 'react';
import renderer from 'react-test-renderer';
import Table from './Table';

const episodesColumns = [
  {
    key: 'episode',
    dataIndex: 'episode',
    title: 'Episode #',
  },
  {
    key: 'title',
    dataIndex: 'title',
    title: 'Title',
  },
  {
    key: 'airDate',
    dataIndex: 'air_date',
    title: 'Air Date',
  },
];

const mockData = [
  {
    air_date: '03-02-2015',
    characters: [
      'Jimmy McGill',
      'Mike Erhmantraut',
      'Kim Wexler',
      'Howard Hamlin',
      'Chuck McGill',
      'Nacho Varga',
    ],
    episode: '5',
    episode_id: 67,
    season: '1',
    series: 'Better Call Saul',
    title: 'Alpine Shepherd Boy',
  },
];

test('Table must render correctly', () => {
  const mockCallBack = jest.fn();

  const component = renderer.create(
    <Table
      columns={episodesColumns}
      data={mockData}
      keyIndex="episode_id"
      onRowCLick={mockCallBack}
    />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
