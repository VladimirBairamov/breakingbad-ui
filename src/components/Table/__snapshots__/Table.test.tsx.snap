// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table must render correctly 1`] = `
<div
  className="flex flex-col"
>
  <div
    className="-my-2 overflow-x-auto sm:-mx-6 lg:-mx-8"
  >
    <div
      className="py-2 align-middle inline-block min-w-full sm:px-6 lg:px-8"
    >
      <div
        className="shadow overflow-hidden border-b border-gray-200 sm:rounded-lg"
      >
        <table
          className="min-w-full divide-y divide-gray-200"
        >
          <thead
            className="bg-gray-50"
          >
            <tr>
              <th
                className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
                scope="col"
              >
                Episode #
              </th>
              <th
                className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
                scope="col"
              >
                Title
              </th>
              <th
                className="px-6 py-3 text-left text-xs font-medium text-gray-500 uppercase tracking-wider"
                scope="col"
              >
                Air Date
              </th>
            </tr>
          </thead>
          <tbody
            className="bg-white divide-y divide-gray-200"
          >
            <tr
              className="cursor-pointer"
              onClick={[Function]}
            >
              <td
                className="px-6 py-4 whitespace-nowrap text-sm text-gray-500"
              >
                5
              </td>
              <td
                className="px-6 py-4 whitespace-nowrap text-sm text-gray-500"
              >
                Alpine Shepherd Boy
              </td>
              <td
                className="px-6 py-4 whitespace-nowrap text-sm text-gray-500"
              >
                03-02-2015
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
